app/components-react/windows/source-showcase/SourceGrid.tsx

```diff
@@ -69,33 +70,125 @@ export default function SourceGrid(p: { activeTab: string }) {
     ? getPlatformService(UserService.state.auth.primaryPlatform)
     : null;
 
-  const iterableWidgetTypes = useMemo(
-    () =>
-      Object.keys(WidgetType)
-        .filter((type: string) => isNaN(Number(type)) && type !== 'SubscriberGoal')
-        .filter((type: string) => {
-          const widgetPlatforms = WidgetDisplayData(primaryPlatform)[WidgetType[type]]?.platforms;
-          if (!widgetPlatforms) return true;
-          return linkedPlatforms?.some(
-            platform => widgetPlatforms && widgetPlatforms.has(platform),
-          );
-        })
-        .filter(type => {
-          // show only supported widgets
-          const whitelist = primaryPlatformService?.widgetsWhitelist;
-          if (!whitelist) return true;
-          return whitelist.includes(WidgetType[type]);
-        }),
-    [],
-  );
+  const iterableWidgetTypesBase = useMemo(() => {
+    const filtered = Object.keys(WidgetType)
+      .filter((type: string) => isNaN(Number(type)) && type !== 'SubscriberGoal')
+      .filter((type: string) => {
+        const widgetPlatforms = WidgetDisplayData(primaryPlatform)[WidgetType[type]]?.platforms;
+        if (!widgetPlatforms) return true;
+        return linkedPlatforms?.some(platform => widgetPlatforms && widgetPlatforms.has(platform));
+      })
+      .filter(type => {
+        // show only supported widgets
+        const whitelist = primaryPlatformService?.widgetsWhitelist;
+        if (!whitelist) return true;
+        return whitelist.includes(WidgetType[type]);
+      });
```

Comment:
Is it possible to do this in one pass with reduce?

Reply:
For performance, yes. For readability IMHO I don't believe the intermediate assignments it would require and the longer code branch in general would provide an improvement, it is also memoized, what do you think?